### DISCUSSION TITLE
Centralized Styles

### DISCUSSION THREAD

**Discussion by m5x5** [Zettlr/Zettlr#2427] [Ideas]
Hello everyone,
I have discovered Zettlr today, and really like the program this far. Because I found some design stuff on Linux, I instantly headed towards creating a new issue, forking the project and helping improve it.

One thing I have noticed during development, is that we have styles for .darwin .win64 and .linux separated, and that there are oftentimes implementation for every single one of them (at least from what I have seen).
What do you think about putting them together and making it one design only for the majority of the project?
I would greatly help to keep an overview of what is going on with all the styles, and would help us to avoid styling it for every OS individually. 🙂

**Comment by nathanlesage** [comment]
After having written quite a lengthy response I have figured out something:

All elements and components in Zettlr have a generic "body" section where styles are being defined irrespective of the platform.

Right now, we additionally have sections for Windows, macOS, and Linux. While we cannot remove the Windows and macOS styles since there we have exactly one GUI and thus can create a native theme, we might be able to merge the "Linux" sections into the generic body section. By doing this, we might be able to simplify the code a little bit, and also make sure that Zettlr looks good on officially unsupported platforms (such as FreeBSD).

So what we might be able to do is the following:

1. First we have a body section that implements a custom design that works on every platform.
2. For Windows and macOS, however, since there is only one set of elements, we can and so must implement a native theme.

Bascially, any Electron app is a desktop app and not a web app and as such it must look native wherever possible. On Windows and macOS it's possible, on Linux it's much more difficult, so this would be a compromise.
